packages/accounts/src/light-account/client.ts

```diff
@@ -18,49 +19,81 @@ import {
 } from "./decorator.js";
 
 export type CreateLightAccountClientParams<
-  TTransport extends Transport = Transport,
+  TEntryPointVersion extends EntryPointVersion,
   TChain extends Chain | undefined = Chain | undefined,
   TSigner extends SmartAccountSigner = SmartAccountSigner
 > = {
-  transport: CreateLightAccountParams<TTransport, TSigner>["transport"];
-  chain: CreateLightAccountParams<TTransport, TSigner>["chain"];
+  transport: CreateLightAccountParams<
+    TEntryPointVersion,
+    Transport,
+    TSigner
+  >["transport"];
+  chain: CreateLightAccountParams<
+    TEntryPointVersion,
+    Transport,
+    TSigner
+  >["chain"];
   account: Omit<
-    CreateLightAccountParams<TTransport, TSigner>,
+    CreateLightAccountParams<TEntryPointVersion, Transport, TSigner>,
     "transport" | "chain"
   >;
 } & Omit<
-  SmartAccountClientConfig<TTransport, TChain>,
+  SmartAccountClientConfig<TEntryPointVersion, Transport, TChain>,
   "transport" | "account" | "chain"
 >;
 
 export function createLightAccountClient<
   TChain extends Chain | undefined = Chain | undefined,
   TSigner extends SmartAccountSigner = SmartAccountSigner
 >(
-  args: CreateLightAccountClientParams<Transport, TChain, TSigner>
+  args: CreateLightAccountClientParams<EntryPointVersion, TChain, TSigner>
 ): Promise<
   SmartAccountClient<
-    CustomTransport,
-    Chain,
-    LightAccount<TSigner>,
-    SmartAccountClientActions<Chain, SmartContractAccount> &
-      LightAccountClientActions<TSigner, LightAccount<TSigner>>
+    EntryPointVersion,
+    Transport,
+    TChain,
+    LightAccount<EntryPointVersion, TSigner>,
+    SmartAccountClientActions<
+      EntryPointVersion,
+      TChain,
+      SmartContractAccount<EntryPointVersion>
+    > &
+      LightAccountClientActions<
+        EntryPointVersion,
+        TSigner,
+        LightAccount<EntryPointVersion, TSigner>
+      >
   >
 >;
 
-export async function createLightAccountClient({
+export async function createLightAccountClient<
+  TChain extends Chain | undefined = Chain | undefined,
+  TSigner extends SmartAccountSigner = SmartAccountSigner
+>({
   account,
   transport,
   chain,
   ...clientConfig
-}: CreateLightAccountClientParams): Promise<SmartAccountClient> {
```

Comment:
we should leave this as it was before without specifying generics, the point of the overload was so that we wouldn't get typing issues when using combinations of chains, signers, or transports